### ISSUE TITLE
Bad cast from type DB::ColumnNullable to DB::ColumnVector<unsigned long>

### ISSUE THREAD

**Issue by dvinegla** [ClickHouse/ClickHouse#39838]
**Describe what's wrong**

If generates a query with:

- SET group_by_use_nulls = 1
- Using GROUPING SETS
- Aliasing count()

i.e.

```sql
SELECT
    fa,
    fb,
    count() AS counter
FROM tb
GROUP BY
    GROUPING SETS (
        (fa),
        (fb))
```

the query fails with bad cast error 

**Does it reproduce on recent release?**

v22.7.1.2484-stable 
clickhouse/clickhouse-server:22.7.1 on Mac m1

**How to reproduce**
```
2f618eb35740 :) CREATE TABLE tb
                (
                    `fa` UInt8,
                    `fb` UInt8
                )
                ENGINE = MergeTree
                ORDER BY tuple()

CREATE TABLE tb
(
    `fa` UInt8,
    `fb` UInt8
)
ENGINE = MergeTree
ORDER BY tuple()

Query id: 30544893-f475-49dc-8cea-8ee3dc2b2733

Ok.

0 rows in set. Elapsed: 0.068 sec.

2f618eb35740 :) INSERT INTO TABLE tb VALUES
                (1, 1)
                (1, 0)
                (0, 0)
                (0, 1)

INSERT INTO tb FORMAT Values

Query id: 7ac661ab-b362-41e3-b582-c6d0cb4150ed

Ok.

4 rows in set. Elapsed: 0.026 sec.

2f618eb35740 :) SELECT
                    fa,
                    fb,
                    count() AS counter
                FROM tb
                GROUP BY
                    GROUPING SETS (
                        (fa),
                        (fb))

SELECT
    fa,
    fb,
    count() AS counter
FROM tb
GROUP BY
    GROUPING SETS (
        (fa),
        (fb))

Query id: cd7fa660-905d-4c39-8df1-e0253c7edacd

┌─fa─┬─fb─┬─counter─┐
│  0 │  0 │       2 │
│  0 │  1 │       2 │
└────┴────┴─────────┘
┌─fa─┬─fb─┬─counter─┐
│  0 │  0 │       2 │
│  1 │  0 │       2 │
└────┴────┴─────────┘

4 rows in set. Elapsed: 0.013 sec.

2f618eb35740 :) SET group_by_use_nulls = 1

SET group_by_use_nulls = 1

Query id: 6cd55174-eb22-48be-abe6-f8878fb6740a

Ok.

0 rows in set. Elapsed: 0.006 sec.

2f618eb35740 :) SELECT
                    fa,
                    fb,
                    count()
                FROM tb
                GROUP BY
                    GROUPING SETS (
                        (fa),
                        (fb))

SELECT
    fa,
    fb,
    count()
FROM tb
GROUP BY
    GROUPING SETS (
        (fa),
        (fb))

Query id: 51c17554-aead-488f-a782-2efa95bc96e9

┌─fa─┬───fb─┬─count()─┐
│  0 │ ᴺᵁᴸᴸ │       2 │
│  1 │ ᴺᵁᴸᴸ │       2 │
└────┴──────┴─────────┘
┌───fa─┬─fb─┬─count()─┐
│ ᴺᵁᴸᴸ │  0 │       2 │
│ ᴺᵁᴸᴸ │  1 │       2 │
└──────┴────┴─────────┘

4 rows in set. Elapsed: 0.004 sec.

2f618eb35740 :) SELECT
                    fa,
                    fb,
                    count() as counter
                FROM tb
                GROUP BY
                    GROUPING SETS (
                        (fa),
                        (fb))

SELECT
    fa,
    fb,
    count() AS counter
FROM tb
GROUP BY
    GROUPING SETS (
        (fa),
        (fb))

Query id: 7d7548de-a353-490a-b0af-05a65b86a0d8


0 rows in set. Elapsed: 0.024 sec.

Received exception from server (version 22.7.1):
Code: 49. DB::Exception: Received from localhost:9000. DB::Exception: Bad cast from type DB::ColumnNullable to DB::ColumnVector<unsigned long>. (LOGICAL_ERROR)

```

**Expected behavior**

> Query should return aggregations.

**Error message and/or stacktrace**

```
Received exception from server (version 22.7.1):
Code: 49. DB::Exception: Received from localhost:9000. DB::Exception: Bad cast from type DB::ColumnNullable to DB::ColumnVector<unsigned long>. (LOGICAL_ERROR)
```

**Comment by dvinegla** [comment]
Also happens with 
22.7.3 revision 54457

**Comment by alexey-milovidov** [comment]
@novikd this bug should have been found by Fuzzer. Let's check why it didn't.
Most likely the fuzzer does not mutate the AST near GROUPING SETS.

**Comment by dvinegla** [comment]
fixed in 22.9.1 revision 54460
